File: Hello LeetCode/LeetCodeBook-Swift/Leetcode-Book/Leetcode-Book/LinkedList/_206_反转链表.swift
//
//  _206_反转链表.swift
//  Leetcode-Book
//
//  Created by gaoliang on 2021/12/9.
//

import Foundation


class _206_反转链表_Solution {
    func reverseList(_ head: ListNode?) -> ListNode? {
        var cur = head
        var prev: ListNode? = nil
        
        while cur != nil {
            let next = cur?.next
            cur?.next = prev
            prev = cur
            cur = next
        }
        
        return prev
    }
}

File: Hello LeetCode/LeetCodeBook-Java/src/com/algorithm/链表/_206_反转链表.java
package com.algorithm.链表;

/*
* https://leetcode-cn.com/problems/reverse-linked-list/
*
* */
public class _206_反转链表 {

    public ListNode reverseList(ListNode head) {
        if (head == null || head.next == null) return head;

        ListNode prev = null;
        ListNode current = head;

        while (current != null) {
            ListNode next = current.next;
            current.next = prev;
            prev = current;
            current = next;
        }

        return prev;
    }
}
